src/promptflow/promptflow/_core/tracer.py

```diff
@@ -168,6 +172,46 @@ def _create_trace_from_function_call(f, *, args=[], kwargs={}, trace_type=TraceT
     )
 
 
+tracer = get_tracer("promptflow")
+
+
+def get_node_name_from_context():
+    tracer = Tracer.active_instance()
+    if tracer is not None:
+        return tracer._node_name
+    return None
+
+
+def enrich_span_with_trace(span, trace):
+    operation_context = OperationContext.get_instance()
+    span.set_attributes(
+        {
+            "framework": "promptflow",
+            "span_type": f"promptflow.{trace.type}",
+            "function": trace.name,
+            "inputs": serialize_attribute(trace.inputs),
+            "node_name": get_node_name_from_context(),
+            "tool_version": "tool_version",  # TODO: Check how to pass the tool version in
+            "flow_id": operation_context.get("flow-id", ""),
+            "root_run_id": operation_context.get("root-run-id", ""),
```

Comment:
Let's sync on these:
![image](https://github.com/microsoft/promptflow/assets/47586720/ca1b3877-fb01-46d9-9f41-cb330f562264)

Reply:
Thank you for the sync. Here's a brief rundown based on our conversation:
- session_id: This is the identifier for the start trace session. It can be set manually using the pf util, or it will be generated automatically if not provided.
- path: This refers to the absolute path of the python file in eager mode or the flow.dag.yaml file when running a DAG. Due to its user-sensitive nature, it's important to note that this should not be included in the OperationContext for tracing purposes.
- flow_id or run: This serves as the identifier to differentiate between individual runs.
- experiment: Currently, this concept has not been fully integrated on the SDK side. We can consider it a placeholder within the attributes for future implementation.

Reply:
is session id and experiment mainly for batch/evaluation?  If so, i'm assuming they are optional and will not be there after deployment, is this correct? @liucheng-ms  @wangchao1230 

"flow_id or run: This serves as the identifier to differentiate between individual runs."
>> seems we don't expose flow_id concept to customer?  can we clarify the difference among root_run_id, flow_id and run?

Reply:
Yes. I think they should be none for deployment scenario.

Reply:
~~`root_run_id`: LEGACY NAME, TO BE REMOVED. We deprecate this name as it does not indicate which level the run is. (It could be both batch run and test run, which is confusing.)
~~`flow_run_id`: LEGACY NAME, TO BE REMOVED. Same as above. 

`batch_run_id`: The batch run id. Can be empty when it is not a batch run.
`line_run_id`: The line run id. It is the run ID for both `flow test` and a line inside a batch run.
`line_no`: An integer starting from 0(?). Only exists when the run is a line inside a batch run.

`referenced.batch_run_id`: When running an evaluation flow, use this attribute to note the batch run id it ran against.
`referenced.line_run_id`: Optional. Set this id when running an evaluation flow against a line.
`referenced.line_no`: Use along with referenced.batch_run_id.